packages/blocks/src/api/utils.js

```diff
@@ -140,6 +141,19 @@ export function normalizeBlockType( blockTypeOrName ) {
  * @return {string} The block label.
  */
 export function getBlockLabel( blockType, attributes, context = 'visual' ) {
+	// Attempt to find entity title if block is a template part.
+	// Require slug to request, otherwise entity is uncreated and will throw 404.
+	if ( 'core/template-part' === blockType.name && attributes.slug ) {
```

Comment:
This logic should be added as `__experimentalLabel` in the `registerBlockType` call for the Template Part block.